src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -103,8 +113,30 @@ protected internal override PinMode GetPinMode(int pinNumber)
 
         protected internal override bool IsPinModeSupported(int pinNumber, PinMode mode)
         {
-            // Libgpiod Api do not support pull up or pull down resistors for now.
-            return mode != PinMode.InputPullDown && mode != PinMode.InputPullUp;
+            switch (mode)
+            {
+                case PinMode.Input:
+                    return true;
+                case PinMode.InputPullDown:
+                case PinMode.InputPullUp:
+                    // for use the bias flags we need Kernel version 5.5 or later
+                    var linuxVersionMatch = Regex.Split(RuntimeInformation.OSDescription, @"Linux (.*?)\.(.*?)\.");
```

Comment:
Would it be possible to try to avoid using Regex in this case and instead use string manipulation? Most of the time using Regex leads to performance bottlenecks and potential security issues, so if we can try using string manipulation instead I think it would be better.

Reply:
Thanks for the review. I fix it on 5344c80